graylog2-web-interface/src/views/components/MigrateFieldCharts.jsx

```diff
@@ -0,0 +1,207 @@
+    case 'monotone':
+      interpolation = 'spline';
+      break;
+    case 'linear':
+    case 'step-after':
+      interpolation = legacyInterpolation;
+      break;
+    default:
+      interpolation = 'linear';
+  }
+
+  switch (visualization) {
+    case 'line':
+      return new LineVisualizationConfig(interpolation);
+    case 'area':
+      return new AreaVisualizationConfig(interpolation);
+    default:
+      return undefined;
+  }
+};
+
+const _initialRowOffset = (widgetPositions) => {
+  const lastWidgetPosition = maxBy(values(widgetPositions), (position: WidgetPosition): number => position.row);
+  return lastWidgetPosition ? (lastWidgetPosition.row + lastWidgetPosition.height) : 0;
+};
+
+const _onMigrate = (legacyCharts: Array<LegacyFieldChart>, setMigrating: boolean => void, setMigrationFinished: boolean => void) => {
+  setMigrating(true);
+  const { defaultHeight } = widgetDefinition('AGGREGATION');
+  const currentView = CurrentViewStateStore.getInitialState();
+  const newWidgetPositions = { ...currentView.state.widgetPositions };
+  const initialRowOffset = _initialRowOffset(newWidgetPositions);
+
+  const newWidgets = legacyCharts.map((chart: LegacyFieldChart, index: number) => {
+    const { field } = chart;
+    // The old field charts only have one series per chart.
+    // The series always relates to the selected field.
+    const series = new Series(mapSeries(chart.valuetype, field));
+    // Because all field charts show the results for the defined timerange,
+    // the new row pivot always contains the timestamp field.
+    const rowPivotConfig = { interval: { type: 'timeunit', ...mapTime(chart.interval) } };
+    const rowPivot = new Pivot('timestamp', 'time', rowPivotConfig);
+    const visualization = mapVisualization(chart.renderer);
+    const visualizationConfig = createVisualizationConfig(chart.interpolation, visualization);
+    // create widget with migrated data
+    const widgetConfig = AggregationWidgetConfig.builder()
+      .visualization(visualization)
+      .visualizationConfig(visualizationConfig)
+      .series([series])
+      .rowPivots([rowPivot])
+      .build();
+    const newWidget = AggregationWidget.builder()
+      .newId()
+      .timerange(undefined)
+      .config(widgetConfig)
+      .build();
+    // create widget position for new widget
+    const migratedWidgetsOffset = defaultHeight * index;
+    const widgetRowPos = initialRowOffset + migratedWidgetsOffset;
+    newWidgetPositions[newWidget.id] = new WidgetPosition(1, widgetRowPos || 1, defaultHeight, Infinity);
+    return newWidget;
+  });
+
+  const newViewState = currentView.state
+    .toBuilder()
+    .widgets(Immutable.List([...currentView.state.widgets, ...newWidgets]))
+    .widgetPositions(newWidgetPositions)
+    .build();
+
+  ViewStatesActions.update(currentView.activeQuery, newViewState).then(
+    () => SearchActions.executeWithCurrentState().then(() => {
+      Store.set('pinned-field-charts-migrated', true);
```

Comment:
Field names like `pinned-field-charts` & `pinned-field-charts-migrated` should be extracted to constants in this module.